### DISCUSSION TITLE
Unable to build version 92 using VS2015

### DISCUSSION THREAD

**Discussion by merajhussain** [cefsharp/CefSharp#3822] [General]
I am trying to build  **cefsharp/92** branch and I get below errors .  I used vs 2015 professional to build the project

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS1617	Invalid option '7.1' for /langversion; must be ISO-1, ISO-2, Default or an integer in range 1 to 6.	CefSharp	C:\CefSharp\CefSharp\CSC	1	Active
Error	CS1617	Invalid option '7.1' for /langversion; must be ISO-1, ISO-2, Default or an integer in range 1 to 6.	CefSharp.Core	C:\CefSharp\CefSharp.Core\CSC	1	Active
Error	CS0006	Metadata file 'C:\CefSharp\CefSharp.BrowserSubprocess.Core\bin\x64\Release\CefSharp.BrowserSubprocess.Core.dll' could not be found	CefSharp.BrowserSubprocess	C:\CefSharp\CefSharp.BrowserSubprocess\CSC	1	Active
Error	CS0006	Metadata file 'C:\CefSharp\CefSharp\bin\Release\CefSharp.dll' could not be found	CefSharp.BrowserSubprocess	C:\CefSharp\CefSharp.BrowserSubprocess\CSC	1	Active
Error	CS1617	Invalid option '7.1' for /langversion; must be ISO-1, ISO-2, Default or an integer in range 1 to 6.	CefSharp.OffScreen	C:\CefSharp\CefSharp.OffScreen\CSC	1	Active
Error	CS1617	Invalid option '7.1' for /langversion; must be ISO-1, ISO-2, Default or an integer in range 1 to 6.	CefSharp.WinForms	C:\CefSharp\CefSharp.WinForms\CSC	1	Active
Error	CS1617	Invalid option '7.1' for /langversion; must be ISO-1, ISO-2, Default or an integer in range 1 to 6.	CefSharp.Wpf	C:\CefSharp\CefSharp.Wpf\CSC	1	Active
Error	CS0006	Metadata file 'C:\CefSharp\CefSharp.Core\bin\Release\CefSharp.Core.dll' could not be found	CefSharp.OffScreen.Example	C:\CefSharp\CefSharp.OffScreen.Example\CSC	1	Active
Error	CS0006	Metadata file 'C:\CefSharp\CefSharp.Core.Runtime\bin\x64\Release\CefSharp.Core.Runtime.dll' could not be found	CefSharp.OffScreen.Example	C:\CefSharp\CefSharp.OffScreen.Example\CSC	1	Active
Error	CS0006	Metadata file 'C:\CefSharp\CefSharp\bin\Release\CefSharp.dll' could not be found	CefSharp.OffScreen.Example	C:\CefSharp\CefSharp.OffScreen.Example\CSC	1	Active
Error	CS0006	Metadata file 'C:\CefSharp\CefSharp.Example\bin\x64\Release\CefSharp.Example.dll' could not be found	CefSharp.OffScreen.Example	C:\CefSharp\CefSharp.OffScreen.Example\CSC	1	Active
Error	CS0006	Metadata file 'C:\CefSharp\CefSharp.OffScreen\bin\Release\CefSharp.OffScreen.dll' could not be found	CefSharp.OffScreen.Example	C:\CefSharp\CefSharp.OffScreen.Example\CSC	1	Active
Error	CS1617	Invalid option '7.3' for /langversion; must be ISO-1, ISO-2, Default or an integer in range 1 to 6.	CefSharp.Test	C:\CefSharp\CefSharp.Test\CSC	1	Active
Error	CS0006	Metadata file 'C:\CefSharp\CefSharp.Core\bin\Release\CefSharp.Core.dll' could not be found	CefSharp.WinForms.Example	C:\CefSharp\CefSharp.WinForms.Example\CSC	1	Active

![image](https://user-images.githubusercontent.com/13115973/134621432-6584f594-6f78-41e2-bf03-58ec977a6877.png)

Can someone help me?

**Comment by amaitland** [comment]
> I used vs 2015 professional to build the project

I'd suggest using `VS2019`.

It's been years since I actually built `CefSharp3.sln` from within the `VS2015 IDE`, it might not actually be possible. Newer language features are used which use the [Microsoft.Net.Compilers](https://github.com/cefsharp/CefSharp/blob/cefsharp/92/CefSharp/packages.config#L4) nuget package

If you absolutely need to use `VS2015` then try building from `powershell`

```ps
.\build.ps1
```

`AppVeyor` created the official builds using `VS2015`, it was done via the powershell script though.

https://github.com/cefsharp/CefSharp/blob/cefsharp/92/appveyor.yml#L20

**Reply by amaitland** [reply]
If your intention is to test out some aspect of the project then I'd suggest using https://github.com/cefsharp/CefSharp.MinimalExample which should work from with `VS2015 IDE`.

**Reply by merajhussain** [reply]
thanks @amaitland  some months ago I was able to build branch 84 with vs 2015 successfully using the IDE. Any way should I use any specific build of vs 2019 or any latest vs2019 is fine?

**Reply by amaitland** [reply]
> some months ago I was able to build branch 84 with vs 2015 successfully using the IDE.

The new compiler was introduced in version `87. See https://github.com/cefsharp/CefSharp/commit/5d4897f4c6193722a20707289b0d9defa71c5a7c#diff-8626d7cdacc2171b58fc8aa6090ac219fa01dd17823348705197469a4d3b1ea8

If the project compiles using `build.ps1` then it might just be that  the package restore needed to happen before `VS2015` is opened. 

> latest vs2019 is fine

Latest should be fine.

**Reply by merajhussain** [reply]
able to build with VS2019
